tool/client.cc

```diff
@@ -540,24 +682,58 @@ bool Client(const std::vector<std::string> &args) {
     SSL_CTX_set_permute_extensions(ctx.get(), 1);
   }
 
+  std::string certPathFlag;
+  int verify = SSL_VERIFY_NONE;
   if (args_map.count("-root-certs") != 0) {
+    certPathFlag = "-root-certs";
+    verify = SSL_VERIFY_PEER;
+  }
+  // For the OpenSSL tool, simply specifying -CAfile does not imply verification
+  if (args_map.count("-CAfile") != 0) {
+    certPathFlag = "-CAfile";
+  }
```

Comment:
* Should we only allow one of the four options (`-root-certs`, `-CAfile`, `-root-cert-dir` or `-CApath`) to be specified in a single invocation, otherwise produce an error? 
  * It seems clear that at least certain combinations should not be allowed b/c they could lead to unexpected (or misleading) outcomes. 
  * For example, `-root-certs` and `-CAfile` would cause the `-root-certs` parameter to be ignored, but our `SSL_VERIFY_PEER` flag would still be set. And the client might interpret the outcome as applying to the `-root-certs` parameter when actually that parameter was ignored.
  * Even if OpenSSL allows certain "invalid" combinations, if the outcome would likely mislead our consumer, we should diverge from OpenSSL's behavior to improve the security.

Reply:
Ah so -root-certs and -root-cert-dir are equivalents to -CAfile and -CApath dir, but they are only specified for the bssl tool. CAfile and CApath are new for our openssl tool (which is different from the bssl tool). In the startup process for the openssl s_client tool, we parse the provided arguments ([link](https://github.com/aws/aws-lc/pull/1959/files/ca4212844eb02c773d8433481242a4d65d1198c1#diff-44dd5728650be0db0986f6c19ace232b650cbe797d21d413c58198d766e9d4d5R34)) and would fail if -root-certs or -root-cert-dir are provided (since they aren't part of the valid argument map for the openssl tool ([link](https://github.com/aws/aws-lc/pull/1959/files/ca4212844eb02c773d8433481242a4d65d1198c1#diff-44dd5728650be0db0986f6c19ace232b650cbe797d21d413c58198d766e9d4d5R9))).

Reply:
Since such combinations of parameters should be impossible, it could be helpful to have some debug assertions in this function to verify this as a precondition.